cpp/src/arrow/extension/fixed_shape_tensor.cc

```diff
@@ -104,6 +104,38 @@ bool FixedShapeTensorType::ExtensionEquals(const ExtensionType& other) const {
          permutation_equivalent;
 }
 
+std::string FixedShapeTensorType::ToString() const {
+  std::stringstream ss;
+  ss << "extension<" << this->extension_name()
+     << "[value_type=" << value_type_->ToString() << ", shape=[";
+  std::string separator;
+  for (auto v : shape_) {
+    ss << separator << v;
```

Comment:
Using an uninitialized variable here [seems to be ok](https://stackoverflow.com/questions/17738439/value-and-size-of-an-uninitialized-stdstring-variable-in-c), but is otherwise probably best avoided.
.

Reply:
Which uninitialized variable?

Reply:
`separator`.

Reply:
`separator` is not uninitialized, its empty constructor is called.

Reply:
Also, rather than StackOverflow, I would recommend the [C++ reference](https://en.cppreference.com/w/cpp/string/basic_string).